sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/udf/BeamBuiltinFunctionClass.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.impl.udf;
+
+/** BeamBuiltinFunctionClass interface. */
+interface BeamBuiltinFunctionClass {}
```

Comment:
I think this should be called `BuiltinFunctionsProvider` so that it doesn't look weird if there are multiple functions implemented here. Plus it somewhat aligns with service loader/provider terminology this way.

I would also make it an abstract class that can extract UDFs from itself:

```
abstract BuiltinFunctionsProvider {

   protected List<UDF> loadUDFs() {
       this.getClass().getDeclaredMethods().stream().filter(BuiltinFunctionsProvider::isUDF).map(BuiltinFunctionsProvider::toUDF).collect(toList());
   }
}
```

Reply:
This is a great idea. I have changed the built-int function registration to service loader/provider  style.